File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/EventProperties.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class EventProperties(
  val anonymousUserID: String,
  val prefix: String,
  val client: String,
  val source: SourceEnum, // Oneof: IDEEXTENSION
) {

  enum class SourceEnum {
    @SerializedName("IDEEXTENSION") IDEEXTENSION,
  }
}

File: agent/bindings/csharp/EventProperties.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class EventProperties
  {
    [JsonProperty(PropertyName = "anonymousUserID")]
    public string AnonymousUserID { get; set; }
    [JsonProperty(PropertyName = "prefix")]
    public string Prefix { get; set; }
    [JsonProperty(PropertyName = "client")]
    public string Client { get; set; }
    [JsonProperty(PropertyName = "source")]
    public SourceEnum Source { get; set; } // Oneof: IDEEXTENSION

    public enum SourceEnum
    {
      [EnumMember(Value = "IDEEXTENSION")]
      IDEEXTENSION,
    }
  }
}
